packages/babel-plugin-proposal-object-rest-spread/test/fixtures/regression/gh-8323/input.js

```diff
@@ -0,0 +1,3 @@
+const f = ({ a, b = 0, ...z }) => {
```

Comment:
I'm wondering about "reversed" case of:
```js
const f = ({ a = 0, b, ...z }) => {
  const v = b + 3;
}
```

Judging based on the code it will remove `a` and I'm not sure if it should. The whole point of this optimization was to output terser code for spread as "omit" case and removing properties which have explicit defaults doesnt seem falling into that use case - my expectancy would be that either it shouldnt have a default value if it was used merely as part of the omit operation OR that it shouldnt be there at all if the identifier is intentionally unused.

WDYT?

Reply:
> you'd have a much better sense of what the right thing to do here is.

Why would that be? 😉 I'm just thinking out loud to get a second opinion. What I'm trying to say - I'm not any better than you 😃 

> I think what you're saying is that if someone went through the trouble of assigning a default value we probably shouldn't remove it from the output?

Yeah.

> presumably if you aren't using it the defualt value was unnecessary.

I just kinda feel that this maybe is a linter's job to warn against such cases

> fireTheMissiles

I'm not that much concern about this case, because the optimization happens only in loose mode anyway & can also "break" other code such as:
```js
const f = ({ a, b = 0, ...z }) => {
  const v = b + 3;
};

f({
  get a() {
    fireTheMissiles()
    return 3
  }
})
```

although the breakage possibility is much higher with functions as default parameters than with getters.

Personally I would leave defaults out of this optimization but I'm not overly attached to it.